test/parallel/test-http-client-reject-unexpected-agent.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const baseOptions = {
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
+};
+
+const failingAgentOptions = [
+  true,
+  'agent',
+  {},
+];
+
+const acceptableAgentOptions = [
+  false,
+  undefined,
+  null,
+  new http.Agent(),
+];
+
+let requestsReceived = 0;
+
+const server = http.createServer((req, res) => {
+  requestsReceived++;
+  res.end();
+});
+
+server.listen(0, baseOptions.host, function() {
+  baseOptions.port = this.address().port;
+  failingAgentOptions.forEach((agent) => {
+    assert.throws(
+      () => http.request(Object.assign(baseOptions, {agent})),
+      (err) => {
+        return err.message === 'Agent option must be an instance of ' +
+        'http.Agent, undefined or false. Received type: ' +
+        `${typeof agent} instead`;
+      },
+      `Expected agent option of ${agent} to be rejected`
+    );
+  });
+
+
+  acceptableAgentOptions.forEach((agent) => {
+    assert.doesNotThrow(
+      () => {
+        const req = http.request(Object.assign(baseOptions, {agent}));
+        req.end();
+      },
+      Error,
+      `this: ${agent}`
+    );
+  });
+
+  const intervalId = setInterval(() => {
```

Comment:
@cjihrig - what is the correct way to .close() the server with this kind of test. Ideally we wouldn't close it just after request x in case for some awful reason x + 1 request would have erroneously occurred which would need to be caught?

I've been looking in some of the other http type tests but all seem to implement something different.